packages/tangle/scheduler.go

```diff
@@ -334,6 +290,23 @@ func (s *Scheduler) unreadyParents(messageID MessageID) (parents []MessageID) {
 	return
 }
 
+// tryReady tries to set the given message as ready.
+func (s *Scheduler) tryReady(messageID MessageID) {
+	if s.isReady(messageID) {
+		if err := s.Ready(messageID); err != nil {
+			s.Events.Error.Trigger(errors.Errorf("failed to mark %s as ready: %w", messageID, err))
+		}
+	}
+}
+
+// updateApprovers iterates over the direct approvers of the given messageID and
+// tries to mark them as ready.
+func (s *Scheduler) updateApprovers(messageID MessageID) {
+	s.tangle.Storage.Approvers(messageID).Consume(func(approver *Approver) {
+		s.tryReady(approver.approverMessageID)
+	})
```

Comment:
maybe the approver has one parent confirmed and the other not confirmed? This is enough gor try ready?

Reply:
tryReady calls isReady that checks ALL of its parents